model/src/pyrenew/process/simplerandomwalk.py

```diff
@@ -61,12 +72,11 @@ def sample(
             With a single array of shape (n_timepoints,).
         """
 
-        if init is None:
-            init = npro.sample(name + "_init", self.error_distribution)
+        init, *_ = self.init_rv.sample(**kwargs)
 
         def transition(x_prev, _):
             # numpydoc ignore=GL08
-            diff = npro.sample(name + "_diffs", self.error_distribution)
+            diff, *_ = self.step_rv.sample(**kwargs)
```

Comment:
```suggestion
            diff, *_ = self.step_rv(**kwargs)
```

Reply:
It should. I admit to having forgotten about the r.v. `__call__` method when I was implementing this.